espressostreamer/espresso_streamer.go

```
@@ -326,29 +337,43 @@ func fetchNextHotshotBlock(
 	nextHotshotBlockNum uint64,
 	parseHotShotPayloadFn func(tx espressoTypes.Bytes) ([]*MessageWithMetadataAndPos, error),
 	namespace uint64,
+	parseTxPayloadRetryBackoff time.Duration,
 ) ([]*MessageWithMetadataAndPos, error) {
 	arbTxns, err := espressoClient.FetchTransactionsInBlock(ctx, nextHotshotBlockNum, namespace)
 	if err != nil {
-		return []*MessageWithMetadataAndPos{}, fmt.Errorf("%w: %w", FailedToFetchTransactionsErr, err)
+		return []*MessageWithMetadataAndPos{}, fmt.Errorf("%w: %w", ErrFailedToFetchTransactions, err)
 	}
 
 	result := []*MessageWithMetadataAndPos{}
 
 	for _, tx := range arbTxns.Transactions {
-		messages, err := parseHotShotPayloadFn(tx)
-		if err != nil {
-			log.Warn("failed to verify espresso transaction", "err", err)
-			continue
+		var retryCount uint64
+		for {
+			messages, err := parseHotShotPayloadFn(tx)
+			if err == nil {
+				result = append(result, messages...)
+				break
+			}
+
+			// If the error is Fatal (e.g., batch poster signature verification failed), we skip to the next transaction.
+			if ErrorIsFatal(err) {
+				log.Warn("Fatal error while parsing payload, skipping transaction", "err", err)
+				break
+			}
+
+			retryCount++
+			log.Warn("Retriable error while parsing payload, will retry", "err", err, "retryCount", retryCount)
+			// In the case of an rpc error, we retry after a delay.
+			time.Sleep(parseTxPayloadRetryBackoff)
```

I don't think we need a for-loop here. I think we can just return error if the error is not `fatal`?

---

Actually my naming was incorrect, it should be persistent instead. Meaning if we retry it will always fail. Originally we should skip the transaction if it's just invalid (e.g., wrong signature, etc.) but you think we should return an error?